backend/src/api/routes/s3-gateway/index.routes.ts

```
@@ -84,7 +85,45 @@ s3GatewayRouter.use((req, res, next) => {
   s3Sigv4Middleware(req, res, next).catch(next);
 });
 
-// 3) Dispatch to the operation handler.
+// 3) Early Content-Length check for body-consuming operations.
+// For aws-chunked streaming uploads the wire Content-Length includes
+// chunk framing overhead; use x-amz-decoded-content-length instead.
+s3GatewayRouter.use((req: Request, res: Response, next) => {
+  const rawDecoded = req.headers['x-amz-decoded-content-length'];
+  const isStreaming =
+    typeof rawDecoded === 'string' && /^\d+$/.test(rawDecoded);
```

ESLint

[error] Delete `⏎···`

(prettier/prettier)